src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -868,6 +872,14 @@ def create_ag_url_path_map(cmd, resource_group_name, application_gateway_name, i
         new_map.default_redirect_configuration = \
             SubResource(id=default_redirect_config) if default_redirect_config else None
 
+    if cmd.supported_api_version(min_api='2019-04-01'):
```

Comment:
again, it looks really bad for adding hardcode versions in different piece of codes but same purpose. at least define something like global var rewrite_rule_set_min_version

Reply:
A possible reason behind hardcode might be if we need to define a global var, the name might be `ag_url_path_map_rewrite_rule_set_min_version`. This api-version is limited in these two commands `az network application-gateway url-path-map create/update`

Reply:
@yungezz Currently, we set `min_api=2019-04-01` in _param.py too. I just checked the structure in `cmd` and we can also get this information. How about we support something like `cmd.support_parameter(rewrite_rule_set)` so that we can remove this hardcode version and keep the version information limited in _param.py

Reply:
It can also work for multi api if I write the check like `if rewrite_rule_set is None`

Reply:
thanks @MyronFanQiu . sounds good to add in cmd. Image one day, you'll need update versoin for a feature like rewrite_rule_set, how it's possible to find out all instance of hardcode version. meanwhile, bad readability.
Eventually, I would like we have this practice tested/verified, and make it into cli command guideline.